dev/com.ibm.ws.concurrent_fat_cdi/test-applications/concurrentCDIApp/src/concurrent/cdi/web/ConcurrentCDIServlet.java

```diff
@@ -357,6 +362,24 @@ public void testAppScopedBeanMethodWithoutAnnotationIsNotAsync() throws Exceptio
         assertEquals(curThreadName, threadNameRef.get());
     }
 
+    /**
+     * The Asynchronous annotation can be placed on a method of a superclass of a CDI managed bean.
+     */
+    @Test
+    public void testAsyncOnSuperclassMethod() throws Exception {
+        CompletableFuture<Long> future = requestScopedBean.getThreadId();
+
+        // Get the result in a way that avoids being able to run inline
+        for (long start = System.nanoTime(); System.nanoTime() - start < TIMEOUT_NS && !future.isDone();)
+            TimeUnit.MILLISECONDS.sleep(200);
+
+        long asyncMethodThreadId = future.getNow(-1l);
+        long curThreadId = Thread.currentThread().getId();
+
+        assertTrue("Asynchronous method must not run on current thread " + curThreadId,
+                   Thread.currentThread().getId() != asyncMethodThreadId);
```

Comment:
You caught this yourself, but as a reminder you could just use curThreadId here instead of Thread.currentThread.getId().

Reply:
Thanks for reminding me of this. I updated the pull to make the improvement.